### PR TITLE
fix(regex): company house validation

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
Company house REGEX is not allowing ingestion of `SC467044`.

## Resolution
* Updated REGEX to allow `SC467044` company house number.
* Updated tests.

### CODE REVIEW

File: gef-ui/server/controllers/companies-house/index.test.js [hunk 1]
```diff
@@ -121,21 +121,21 @@ describe('controllers/about-exporter', () => {
     });
 
     it('redirects user to `exporters address` page if response from api is successful', async () => {
-      mockRequest.body.regNumber = '123';
+      mockRequest.body.regNumber = '8989898';
```
Comment (abhi-markan) [inline gef-ui/server/controllers/companies-house/index.test.js:124]: Constant please.

File: gef-ui/server/utils/validateIds.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const { isValidRegex } = require('./validateIds');
+const { COMPANIES_HOUSE_NUMBER_REGEX } = require('../constants');
+
+describe('validateIds', () => {
+  describe('isValidRegex', () => {
+    it('should return true for company number 8989898', () => {
+      const result = isValidRegex(COMPANIES_HOUSE_NUMBER_REGEX, '8989898');
```
Comment (abhi-markan) [inline gef-ui/server/utils/validateIds.test.js:7]: Constants please.

File: portal-api/api-tests/v1/validation/validateIds.api-test.js [hunk 3]
```diff
@@ -35,37 +36,61 @@ describe('validateIds', () => {
...
-      const result = isNotValidCompaniesHouseNumber('8989898');
+      it('should return true for company number RS00592C', () => {
+        const result = isValidRegex(COMPANIES_HOUSE_NUMBER, 'RS00592C');
 
-      expect(result).toEqual(false);
+        expect(result).toEqual(true);
+      });
+
+      it('should return true for company number NI614169', () => {
+        const result = isValidRegex(COMPANIES_HOUSE_NUMBER, 'NI614169');
```
Comment (abhi-markan) [inline portal-api/api-tests/v1/validation/validateIds.api-test.js:60]: Constant please.

File: portal-api/src/v1/validation/validateIds.js [hunk 4]
```diff
@@ -28,25 +27,8 @@ const isValidRegex = (regex, value) => regex.test(value);
...
 const isValidCurrencyCode = (code) => (code ? validator.isISO4217(String(code)) : false);
 
-/**
- * isNotValidCompaniesHouseNumber
- * checks if companiesHouseNumber conforms to schema
- * returns true if validation error or false if not
- * @param {String} companiesHouseNumber
- * @returns {Boolean}
- */
-const isNotValidCompaniesHouseNumber = (companiesHouseNumber) => {
```
Comment (abhi-markan) [inline portal-api/src/v1/validation/validateIds.js:38]: Can we leave the function in please with call to `isValidRegex` and make calls to this functions from tests please?

File: portal/server/companies-house-api.js [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
+const { isValidRegex } = require('./validation/validate-ids');
+const { COMPANIES_HOUSE_NUMBER } = require('./constants/regex');
 
 const urlRoot = process.env.COMPANIES_HOUSE_API_URL;
 const apiKey = process.env.COMPANIES_HOUSE_API_KEY;
 
 const getByRegistrationNumber = async (registrationNumber) => {
   try {
-    if (isNotValidCompaniesHouseNumber(registrationNumber)) {
+    if (!isValidRegex(COMPANIES_HOUSE_NUMBER, registrationNumber)) {
```
Comment (abhi-markan) [inline portal/server/companies-house-api.js:12]: Call to the function please?

File: portal/server/validation/validate-ids.js [hunk 6]
```diff
@@ -11,22 +10,15 @@ const joi = require('joi');
...
- * returns true if validation error or false if not
- * @param {String} companiesHouseNumber
- * @returns {Boolean} Ascertain whether companiesHouseNumber is valid or not.
+ * isValidRegex
+ * validates value conforms to passed regex rules
+ * @param {String} regex
+ * @param {String} value
+ * @returns {Boolean} asserts if regex is matched or not
  */
-const isNotValidCompaniesHouseNumber = (companiesHouseNumber) => {
```
Comment (abhi-markan) [inline portal/server/validation/validate-ids.js:20]: Revert function which calls `isValidRegex` please?

File: portal/server/validation/validate-ids.test.js [hunk 7]
```diff
@@ -45,47 +46,47 @@ describe('validate-ids', () => {
     });
   });
 
-  describe('isNotValidCompaniesHouseNumber', () => {
-    it('should return false if input is a valid companies house number', () => {
-      const result = isNotValidCompaniesHouseNumber('8989898');
+  describe('isValidRegex', () => {
```
Comment (abhi-markan) [inline portal/server/validation/validate-ids.test.js:49]: Updated tests post function name update.

**Review by abhi-markan — COMMENTED** [review]
Few comments to address please.

File: external-api/src/constants/regex.constants.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
-export const regexes = {
+export const REGEX = {
   POSTCODE: /^[A-Za-z]{1,2}[0-9Rr][0-9A-Za-z]?[0-9][ABD-HJLNP-UW-Zabd-hjlnp-uw-z]{2}$/,
-  COMPANIES_HOUSE_NUMBER: /^\d{8,10}$/,
+  COMPANIES_HOUSE_NUMBER_REGEX: /^[0-9A-Z]{6,8}$/,
```
Comment (abhi-markan) [inline external-api/src/constants/regex.constants.ts:3]: Why is the company regex not same as `/^(([A-Z]{2}|[A-Z]\d{1}|\d{2})(\d{5,6}|\d{4,5}[A-Z]))$/;`?

**Review by abhi-markan — COMMENTED** [review]
One comment to address please.

File: gef-ui/server/controllers/companies-house/index.test.js (hunk 1 above)
Comment (abhi-markan) [reply gef-ui/server/controllers/companies-house/index.test.js]: Have added and replaced other test cases as well.

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**